### ISSUE TITLE
Pasta Networking - OUTBOUND Traffic does NOT work, INBOUND Traffic works

### ISSUE THREAD

**Issue by luckylinux** [containers/podman#23003]
### Issue Description

When using `pasta` Networking (at least using IPv6 Address Only) the following can be observed:
- OUTBOUND Traffic does NOT work
- INBOUND Traffic works (using Port Publishing / Mapping)

### Steps to reproduce the issue

Steps to reproduce the issue
1. Patch Pasta According to my Comment in https://github.com/containers/podman/issues/22824 (otherwise it ABSOLUTELY will NOT work)
2. Start Podman Container (the minimal example Provided at https://passt.top/passt/about/#pasta_2 results in a timeout as well: `podman run --net=pasta --rm -ti alpine sh`)

Otherwise for my Debugging Container: https://github.com/luckylinux/docker-network-troubleshooting/

Which is first build then run using (tried several options, OUTBOUND traffic fails with each and every single one of them **except** using `slirp4netns` instead of `pasta`):
```
#!/bin/bash

# Determine toolpath if not set already
relativepath="./" # Define relative path to go from this script to the root level of the tool
if [[ ! -v toolpath ]]; then scriptpath=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd ); toolpath=$(realpath --canonicalize-missing ${scriptpath}/${relativepath}); fi

# Load Configuration
libpath=$(readlink --canonicalize-missing "${toolpath}/includes")
source ${libpath}/functions.sh


# Optional argument
engine=${1-"podman"}

# Container Name
containername="docker-network-troubleshooting"

# Container Image
containerimage="docker-network-troubleshooting:debian-latest"

# Containers Configuration Folder to Map
containersconfigfolder="./containers"

# Options
opts=()

# Run in the Background
opts+=("-d")

# Podman 5.x with Pasta doesn't handle Networking Correctly
# Force to use slirp4netns
#opts+=("--network=slirp4netns")

# Debug Pasta
timestamp=$(date +"%Y%m%d_%Hh%M%S")
#opts+=("--network=pasta:--pcap,./tmp/${timestamp}.pcap")
#opts+=("--network=pasta:--ipv6-only,-t,2XXX:XXXX:XXXX:ff15::2/80,-t,2XXX:XXXX:XXXX:ff15::2/443,-t,2XXX:XXXX:XXXX:ff15::2/5001,-u,2XXX:XXXX:XXXX:ff15::2/5001")
#opts+=("--network=pasta:--ipv6-only,--outbound,2XXX:XXXX:XXXX:ff15::2")
#opts+=("--network=pasta:--ipv6-only,-a,2XXX:XXXX:XXXX:ff15::2")
opts+=("--network=pasta:--ipv6-only,-a,2XXX:XXXX:XXXX:ff15::2,--pcap,./tmp/${timestamp}.pcap")

# Publish ports
opts+=("-p")
opts+=("[2XXX:XXXX:XXXX:ff15::2]:80:80/tcp")
opts+=("-p")
opts+=("[2XXX:XXXX:XXXX:ff15::2]:443:443/tcp")
opts+=("-p")
opts+=("[2XXX:XXXX:XXXX::2]:5201:5201/tcp")
opts+=("-p")
opts+=("[2XXX:XXXX:XXXX:XXXX::2]:5201:5201/udp")

# Add Capacilities
opts+=("--cap-add")
opts+=("CAP_NET_RAW")

# Enable Infinite Loop
opts+=("-e")
opts+=("ENABLE_INFINITE_LOOP=true")

# Disable Automatic Test
opts+=("-e")
opts+=("ENABLE_AUTOMATIC_TEST=false")

# Load the Environment Variables into THIS Script
if [[ -f "./.env" ]]
then
   eval "$(shdotenv --env .env || echo \"exit $?\")"
fi

# Terminate and Remove Existing Containers if Any
${engine} stop --ignore ${containername}
${engine} rm --ignore ${containername}

# Run Image with Infinite Loop to prevent it from automatically terminating
#${engine} run --name=${containername} --env-file "./.env" localhost:5000/local/"${containerimage}"
echo "Executing: ${engine} run --name=${containername} ${opts[*]} localhost:5000/local/${containerimage}"
${engine} run --replace --name=${containername} ${opts[*]} localhost:5000/local/"${containerimage}"

# Open Interactive Shell with Container
#${engine} exec -it ${containername} /bin/bash

```

4. Try to perform a TCP Connection: `curl -6 ifconfig.me` for a Remote Server or `curl -6 https://[2XXX:XXXX:XXXX::0:1]` for my Local OPNSense Router Web Interface

Timeout will be reached.

### Describe the results you received

Timeout reached when running the above mentioned `curl` Commands.

Furthermore:
- `ping -6` (ICMPv6) of LAN Hosts from WITHIN the Container (even on a different Subnet) work correctly
- `ping -6` (ICMPv6) of WAN Hosts from WITHIN the Container Fails
- `traceroute6` work for LAN Hosts (TCP) BUT it takes around 30 seconds, while the reported TTL is shown as ~ 1.7 ms
- `traceroute6` fails for WAN Hosts (both ICMPv6 or TCP)
- Analysis of Wireshark PCAP Shows:
  - DNS Fails (fallback to <requestedhostname>.<MYDOMAIN.TLD>, where search = <MYDOMAIN.TLD>)
  - ICMPv6 seems to work correctly
  - TCP Flags: Lines are either RED (for RST Flag) or BLACK (for SYN Flag & TCP Port Number Reused)
     - RST Flag: present
     - SYN Flag: present
  - TCP Flags missing: Notably absent are ACK or FIN (among others), so the Connection initialized from within the container never gets answered by the outside Services

### Describe the results you expected

Able to reach outside Services normally.

- `curl -6 ifconfig.me` should succeed
- DNS should resolve correctly
- Package Managers (`apk`, `apt`, etc) should be able to download Lists & Updates (when building Containers)

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.36.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.fc40.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: '
  cpuUtilization:
    idlePercent: 99.26
    systemPercent: 0.41
    userPercent: 0.34
  cpus: 8
  databaseBackend: sqlite
  distribution:
    distribution: fedora
    variant: server
    version: "40"
  eventLogger: journald
  freeLocks: 2025
  hostname: podmanserver15.MYDOMAIN.TLD
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1002
      size: 1
    - container_id: 1
      host_id: 655360
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1002
      size: 1
    - container_id: 1
      host_id: 655360
      size: 65536
  kernel: 6.8.11-300.fc40.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 2777296896
  memTotal: 8171499520
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.10.0-1.fc40.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.10.0
    package: netavark-1.10.3-3.fc40.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.10.3
  ociRuntime:
    name: crun
    package: crun-1.15-1.fc40.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.15
      commit: e6eacaf4034e84185fd8780ac9262bbf57082278
      rundir: /run/user/1002/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/local/bin/pasta
    package: Unknown
    version: |
      pasta 2024_06_07.8a83b53-9-g4070bac
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1002/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.2-2.fc40.x86_64
    version: |-
      slirp4netns version 1.2.2
      commit: 0ee2d87523e906518d34a6b423271e4826f71faf
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.5
  swapFree: 10656407552
  swapTotal: 10733215744
  uptime: 30h 35m 48.00s (Approximately 1.25 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  localhost:5000:
    Blocked: false
    Insecure: true
    Location: localhost:5000
    MirrorByDigestOnly: false
    Mirrors: null
    Prefix: localhost:5000
    PullFromMirror: ""
  search:
  - docker.MYDOMAIN.TLD
  - docker.io
  - ghcr.io
store:
  configFile: /home/podman/.config/containers/storage.conf
  containerStore:
    number: 8
    paused: 0
    running: 8
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.13-1.fc40.x86_64
      Version: |-
        fusermount3 version: 3.16.2
        fuse-overlayfs: version 1.13-dev
        FUSE library version 3.16.2
        using FUSE kernel interface version 7.38
    overlay.mountopt: nodev
  graphRoot: /zdata/PODMAN/STORAGE
  graphRootAllocated: 298237820928
  graphRootUsed: 3645636608
  graphStatus:
    Backing Filesystem: zfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /home/podman/containers/tmp
  imageStore:
    number: 222
  runRoot: /run/user/1002
  transientStore: false
  volumePath: /zdata/PODMAN/VOLUMES
version:
  APIVersion: 5.1.0
  Built: 1716940800
  BuiltTime: Wed May 29 02:00:00 2024
  GitCommit: ""
  GoVersion: go1.22.3
  Os: linux
  OsArch: linux/amd64
  Version: 5.1.0
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Running on Fedora 40 in a KVM AMD64 Virtual Machine (Proxmox VE).

### Additional information

Routes from within the Container:
```
root@CONTAINER:/opt/app# ip -6 route
64:ff9b:1::1:0:0/96 via 2XXX:XXXX:XXXX:1::1:66 dev ens18 metric 1024 pref medium
64:ff9b:1::2:0:0/96 via 2XXX:XXXX:XXXX:1::1:67 dev ens18 metric 1024 pref medium
2XXX:XXXX:XXXX:1::/64 dev ens18 proto ra metric 1002 mtu 1500 pref medium
2XXX:XXXX:XXXX:1::/64 via 2XXX:XXXX:XXXX:1::1 dev ens18 metric 1024 pref medium
2XXX:XXXX:XXXX:ff15::/64 dev ens18 proto kernel metric 256 pref medium
fe80::/64 dev ens18 proto kernel metric 256 pref medium
default via fe80::XXXX:XXXX:XXXX:XXXX dev ens18 proto ra metric 1002 mtu 1500 pref medium

root@CONTAINER:/opt/app# ip -6 route show table local
local ::1 dev lo proto kernel metric 0 pref medium
local 2XXX:XXXX:XXXX:ff15::2 dev ens18 proto kernel metric 0 pref medium
local fe80::XXXX:XXXX:XXXX:XXXX dev ens18 proto kernel metric 0 pref medium
multicast ff00::/8 dev ens18 proto kernel metric 256 pref medium
```

Addresses from within the Container:
```
root@CONTAINER:/opt/app# ip -6 addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 state UNKNOWN qlen 1000
    inet6 ::1/128 scope host proto kernel_lo 
       valid_lft forever preferred_lft forever
2: ens18: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 65520 state UNKNOWN qlen 1000
    inet6 2XXX:XXXX:XXXX:ff15::2/64 scope global nodad 
       valid_lft forever preferred_lft forever
    inet6 fe80::XXXX:XXXX:XXXX:XXXX/64 scope link proto kernel_ll 
       valid_lft forever preferred_lft forever
```

Basically, the only way to get OUTBOUND connectivity is to switch to `slirp4netns`.

This affects both `podman build` and `podman run` (probably also `buildah`).

Further Information available in: https://github.com/containers/podman/issues/22824

**Comment by luckylinux** [comment]
Further Info:

```
# On the Host
podman@MYHOST:~$ pasta --config-net --trace -l 20240614_22h27_ALPINE_PASST_TRACE.log
==========================================================================
pasta 2024_06_07.8a83b53-9-g4070bac: /usr/local/bin/passt.avx2 (431421)
0.8448: info:    Template interface: ens18 (IPv4), ens18 (IPv6)
0.8449: info:    Namespace interface: ens18
0.8449: info:    MAC:
0.8449: info:        host: 12:34:56:12:34:56
0.8449: info:    DHCP:
0.8449: info:        assign: 192.168.8.15
0.8449: info:        mask: 255.255.240.0
0.8449: info:        router: 192.168.1.1
0.8449: info:    DNS:
0.8449: info:        192.168.1.1
0.8449: info:    DNS search list:
0.8449: info:        MYDOMAIN.TLD
0.8450: info:    NDP/DHCPv6:
0.8450: info:        assign: 2XXX:XXXX:XXXX:1::8:15
0.8450: info:        router: fe80::YYYY:YYYY:YYYY:YYYY
0.8450: info:        our link-local: fe80::XXXX:XXXX:XXXX:XXXX
0.8450: info:    DNS search list:
0.8450: info:        MYDOMAIN.TLD
0.8788:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
0.8789:          tap: protocol 58, :: -> ff02::16 (1 packet)
0.9797:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
0.9798:          tap: protocol 58, :: -> ff02::16 (1 packet)
1.6987:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
2.7063:          TCP (spliced): cannot set pool pipe size to 8192
2.7063:          TCP (spliced): cannot set pool pipe size to 8192
2.7063:          TCP (spliced): cannot set pool pipe size to 8192
2.7063:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7064:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7065:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7066:          TCP (spliced): cannot set pool pipe size to 8192
2.7228:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
2.7229:          tap: protocol 58, fe80::b0c9:cff:fea8:e565 -> ff02::16 (1 packet)
2.8348:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
2.8349:          tap: protocol 58, fe80::b0c9:cff:fea8:e565 -> ff02::16 (1 packet)
6.6918:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
14.8283:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
30.6908:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
64.9949:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
130.5308:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
261.6029:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)
519.6509:          pasta: epoll event on /dev/net/tun device 17 (events: 0x00000001)

==========================================================================
# Inside the Container

/ # apk update
fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/main/x86_64/APKINDEX.tar.gz
^C/ # ^C

/ # ^C

/ # apk update
fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/main/x86_64/APKINDEX.tar.gz
^C/ # ^C

/ # 
/ # apk update
fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/main/x86_64/APKINDEX.tar.gz
WARNING: updating and opening https://dl-cdn.alpinelinux.org/alpine/v3.20/main: operation timed out
fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/community/x86_64/APKINDEX.tar.gz
```

**Comment by sbrivio-rh** [comment]
This:

> ```
> 1283.7822:          Flow 0 (TCP connection): TAP_SYN_RCVD: CLOSED -> SYN_SENT
> 1283.7822:          Flow 0 (TCP connection): hash table insert: sock 654, bucket: 14659
> 1283.7823:          Flow 0 (TCP connection): TCP reset at tcp_conn_from_tap:1730
> 1283.7823:          Flow 0 (TCP connection): event at tcp_rst_do:1379
> 1283.7823:          Flow 0 (TCP connection): CLOSED: SYN_SENT -> CLOSED
> ```

is pasta checking:

```
	if (!bind(s, sa, sl)) {
		tcp_rst(c, conn);	/* Nobody is listening then */
		goto cancel;
	}
```

if it can bind() to address and port used as destination for the connection. If it succeeds, then it concludes that the address is local, and no other process is listening on that address and port, so it can skip calling connect() and simply reset the connection attempt.

Except that in this case, non-local binds (`net.ipv4.ip_nonlocal_bind` and `net.ipv6.ip_nonlocal_bind`) were enabled (that's not the default). If the address is non-local, pasta's bind() succeeds, and we just reset the connection for anything non-local.

Workaround: either disable non-local binds, or disable the if (!bind(s, sa, sl)) ... check in tcp.c. I'll look for a better solution. We can't just check the sysctls at initialisation time because they might change at runtime, and checking them on every connection would cause significant overhead.

**Comment by luckylinux** [comment]
Setting `sysctl -w net.ipv6.ip_nonlocal_bind=0` "fixes" (bypasses) the Issue, but I would really like to keep non-local binds, otherwise I would have to setup each IP Address for each Container on the Host Interface.

Using `sysctl -w net.ipv6.ip_nonlocal_bind=1` allows the Podman Container to setup its own IP Address without additional work.

The Proposed quick patch by @sbrivio-rh allow to have non-local bind and OUTBOUND connectivity working:

Replace `passt` git source code of `tcp.c` at line 1729
```
if (!bind(s, sa, sl)) {
```

With
```
if (0 && !bind(s, sa, sl)) {
```

**Comment by luckylinux** [comment]
It would actually be interesting to see if somebody else can reproduce, preferably:
a. Using Fedora 40 AMD64
b. Using the double/triple patched pasta version described here and in the associate BUG Report
c. `sysctl -w net.ipv6.ip_nonlocal_bind=1`
d. Try with: `podman run --net=pasta:--ipv6-only --rm -ti alpine sh`
   The --ipv6-only seems key as with `--net=pasta` then `apk update` works (it's just using the ipv4 connectivity)
e. Interface managed by `NetworkManager` (hence a. Fedora) probably due to `noprefixroute`

**Comment by luckylinux** [comment]
So the issue appears to be UDP-only, now that the patches described have been applied.

From within the Container TCP works flawlessly:
```
# Must use the IP associated with ifconfig.me since DNS Resolution does NOT work
root@CONTAINER:/opt/app# curl -vvv -6 -L https://[2600:1901:0:bbc3::]
*   Trying [2600:1901:0:bbc3::]:443...
* Connected to 2600:1901:0:bbc3:: (2600:1901:0:bbc3::) port 443
* ALPN: curl offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384 / x25519 / RSASSA-PSS
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=ho.st
*  start date: Jun  5 15:45:21 2024 GMT
*  expire date: Sep  3 15:45:20 2024 GMT
*  subjectAltName does not match ipv6 address 2600:1901:0:bbc3::
* SSL: no alternative certificate subject name matches target ipv6 address '2600:1901:0:bbc3::'
* Closing connection
* TLSv1.3 (OUT), TLS alert, close notify (256):
curl: (60) SSL: no alternative certificate subject name matches target ipv6 address '2600:1901:0:bbc3::'
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

But of course UDP and DNS Resolution fails:
```
root@CONTAINER:/opt/app# nslookup google.com
;; communications error to 2XXX:XXXX:XXXX:1::1:3#53: timed out
;; communications error to 2XXX:XXXX:XXXX:1::1:3#53: timed out
;; communications error to 2XXX:XXXX:XXXX:1::1:3#53: timed out
;; communications error to 2XXX:XXXX:XXXX:1::1:4#53: timed out
;; communications error to 2XXX:XXXX:XXXX:1::1:5#53: timed out
;; no servers could be reached
```

**Comment by luckylinux** [comment]
I don't really have a better way to reproduce. Alpine doesn't have `curl` where you can specify IPv6 addresses (tried with [], without, with quotes, escaping \[, \], etc, nothing worked).

Simplified `Dockerfile`:
```
FROM debian:testing

ARG TARGETPLATFORM

# Update Sources
# Install other Dependencies / recommended Packages
RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked \
    --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked \
    apt-get update && \
    apt-get install --no-install-recommends -y ca-certificates bash curl wget jq net-tools iproute2 bind9-dnsutils traceroute tcptraceroute iputils-ping iperf3 && \
    echo "Finished installing base Packages"

# Change Shell
RUN chsh -s /bin/bash root
RUN export SHELL="/bin/bash"
RUN ln -sf /bin/bash /bin/sh

# set ENV to execute startup scripts
ENV ENV /etc/profile

# Copy and Execute Entrypoint Script
COPY docker-entrypoint.sh /opt/
RUN chmod +x /opt/docker-entrypoint.sh
ENTRYPOINT ["/opt/docker-entrypoint.sh"]
```

Simplified `docker-entrypoint.sh`
```
#!/bin/bash

# Start the main process and save its PID
# Use exec to replace the shell script process with the main process
exec /bin/bash -c "while true; do sleep 5; done" &
pidh=$!

# Trap the SIGTERM signal and forward it to the main process
trap 'kill -SIGTERM $pidh; wait $pidh' SIGTERM

# Wait for the main process to complete
wait $pidh
```

Generate Timestamp:
```
timestamp=$(date +%Y%m%d)
```

Build:
```
podman build --network=slirp4netns -t "network-debug-minimal:${timestamp}" -f Dockerfile .
```
(I must use `--network=slirp4netns` otherwise even the build will fail since I cannot even download packages ...)

Run:
```
podman run --net=pasta:--ipv6-only -d --name network-debug-minimal localhost/network-debug-minimal:${timestamp}
```

Enter Environment:
```
podman exec -it network-debug-minimal /bin/bash
```

**Comment by luckylinux** [comment]
Looking at Wireshark Display of the captured PCAP file brings more questions than answers.

For `--net=pasta:--ipv6-only`, `nslookup google.com` only attempts to fetch `A` Records (IPv4).
Using `dig AAAA google.com` forces `dig` to **attempt** to fetch `AAAA` Records as it should be.

Nevertheless, UDP Communication with Nameserver is never answered and no IP Addesses are returned.

**Comment by luckylinux** [comment]
OK the UDP Communication error was NOT an issue related to Podman.

It was a wrong setup of the Bind / Named DNS Nameserver.

I had
```
// Unencrypted
listen-on-v6 port 53 { ::1; };
listen-on port 53 { localhost; 127.0.0.1; 192.168.1.3; any; };

// DNS over TLS
listen-on-v6 port 853 tls letsencrypt-tls { ::1; };
listen-on port 853 tls letsencrypt-tls { localhost; 127.0.0.1; 192.168.1.3; any; };
```

If you change the ipv6 to listen to `any` then it works ...
```
// Unencrypted
listen-on-v6 port 53 { ::1; any; };
listen-on port 53 { localhost; 127.0.0.1; 192.168.1.3; any; };

// DNS over TLS
listen-on-v6 port 853 tls letsencrypt-tls { ::1; any; };
listen-on port 853 tls letsencrypt-tls { localhost; 127.0.0.1; 192.168.1.3; any; };

```

**Comment by sbrivio-rh** [comment]
Fixed in [2024_06_24.1ee2eca](https://archives.passt.top/passt-user/20240624210651.61ce77af@elisabeth/), and matching Fedora 40 [update](https://bodhi.fedoraproject.org/updates/FEDORA-2024-4632bfa865).
